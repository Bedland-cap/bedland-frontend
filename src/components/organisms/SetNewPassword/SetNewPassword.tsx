import { useContext } from 'react';

import RulesDescription from 'components/molecules/PasswordRulesDescription/RulesDescription';
import ResetPasswordForm from 'components/organisms/ResetPasswordForm/ResetPasswordForm';
import { ThemeContext } from 'theme/ThemeContext';
import SetNewPasswordContainer, {
  ResPassCenteredHeading,
} from './SetNewPassword.styled';

const SetNewPassword = () => {
  const { palette } = useContext(ThemeContext);

  return (
    <SetNewPasswordContainer palette={palette}>
      <ResPassCenteredHeading>Reset your password</ResPassCenteredHeading>
      <RulesDescription />
      <ResetPasswordForm />
    </SetNewPasswordContainer>
  );
};
export default SetNewPassword;
