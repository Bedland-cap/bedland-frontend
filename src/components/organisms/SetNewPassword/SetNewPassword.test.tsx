import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import ThemeProvider from 'theme/ThemeContext';
import SetNewPassword from './SetNewPassword';

describe('SetNewPassword', () => {
  it('renders correctly', () => {
    const snapshot = renderer
      .create(
        <ThemeProvider>
          <BrowserRouter>
            <SetNewPassword />
          </BrowserRouter>
        </ThemeProvider>,
      )
      .toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
