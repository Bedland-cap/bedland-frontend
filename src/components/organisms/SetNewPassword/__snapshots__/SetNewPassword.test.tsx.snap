// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SetNewPassword renders correctly 1`] = `
.c2 {
  padding: 0rem 2px 0rem 2.9375rem;
  width: 100%;
}

.c3 {
  liststyletype: disc;
  margin-left: 10px;
  padding: 0 0 0 1rem;
}

.c8 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 2.875rem;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  width: 100%;
}

.c7 {
  border: 1px solid #909AFF;
  border-radius: 0.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 2.875rem;
  padding: 0.625rem 1rem 0.625rem 1rem;
  width: 19.875rem;
}

.c7:disabled,
.c7:hover,
.c7:active,
.c7:focus {
  outline: none;
}

.c6 {
  color: white;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 16;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.6875rem;
  width: 19.875rem;
}

.c9 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #EF6517;
  border: 1px solid black;
  border-radius: 0.5rem;
  color: #FCFEFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 2.5rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.625rem 1rem 0.625rem 1rem;
  width: 19.875rem;
}

.c9:disabled,
.c9:hover,
.c9:active,
.c9:focus {
  outline: none;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1rem;
  padding: 0 2.84375rem 0rem 2.84375rem;
  width: 100%;
}

.c0 {
  color: #FCFEFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1.5rem;
  height: 27.6875rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 25.625rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div
  className="c0"
>
  <h1
    className="c1"
  >
    Reset your password
  </h1>
  <div
    className="c2"
  >
    <span>
       Remember, your password must include:
    </span>
     
    <ol
      className="c3"
      style={
        {
          "listStyleType": "disc",
        }
      }
    >
      <li>
        8-24 characters
      </li>
      <li>
        At least one uppercase character
      </li>
      <li>
        At least one special character
      </li>
    </ol>
  </div>
  <form
    className="c4"
    onSubmit={[Function]}
  >
    <div
      className="c5"
    >
      <label
        className="c6"
        htmlFor="password"
        size="medium"
      >
        Your new password
      </label>
      <input
        className="c7"
        id="password"
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Enter your new password"
        type="password"
      />
    </div>
    <div
      className="c5"
    >
      <label
        className="c6"
        htmlFor="confirmPassword"
        size="medium"
      >
        Confirm the password
      </label>
      <input
        className="c7"
        id="confirmPassword"
        name="confirmPassword"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Confirm the password"
        type="password"
      />
    </div>
    <span
      className="c8"
    />
    <button
      className="c9"
      type="submit"
    >
      Reset my password
    </button>
  </form>
</div>
`;
